sources/ecs-gpu-init/cmd/ecs-gpu-init/main.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
+)
+
+const (
+	// NvidiaGPUInfoFilePath is the file path where GPUs and driver info are saved
+	NvidiaGPUInfoFilePath = "/var/lib/ecs/gpu/nvidia-gpu-info.json"
+)
+
+func main() {
+	if err := generateGPUInfoConfig(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+// GenerateGPUInfoConfig generates the configuration used by the ECS agent
+func generateGPUInfoConfig() error {
+	if ret := nvml.Init(); ret != nvml.SUCCESS {
+		return errors.New("Failed to initialize NVML")
+	}
+	defer nvml.Shutdown()
+	version, ret := nvml.SystemGetDriverVersion()
+	if ret != nvml.SUCCESS {
+		return errors.New("Failed to get version")
+	}
```

Comment:
nit: if we logged the value of `ret` then that might help us reason about an error message